app/models/warning.rb

```diff
@@ -6,7 +6,7 @@ def initialize(violation)
       end
 
       def message
-        [help_text, target_nodes].join(" - ")
+        [help_text, target_nodes.join(", ")].join(" - ")
```

Comment:
Could you name these join segments as constants? Maybe `NODE_JOIN_SEGMENT` and `MESSAGE_JOIN_SEGMENT`?